app/helpers/pagelets_helper.rb

```diff
@@ -1,6 +1,10 @@
 module PageletsHelper
   def pagelets_for(mountpoint)
-    Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{action_name}", mountpoint)
+    Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{partial_name}", mountpoint)
+  end
+
+  def partial_name
+    caller_locations.select { |c| c.to_s.match /views/ }.first.path.split('/').last.split('.').first rescue nil
```

Comment:
Can you use a class variable `@virtual_path`?

```
From: /home/shim/Documents/foreman/foreman/app/helpers/pagelets_helper.rb @ line 4 PageletsHelper#pagelets_for:

    2: def pagelets_for(mountpoint)
    3:   binding.pry
 => 4:   Pagelets::Manager.sorted_pagelets_at("#{controller_name}/#{action_name}", mountpoint)
    5: end

[1] pry(#<#<Class:0x007f97f4728358>>)> @virtual_path
=> "hosts/_form"
```

According to [this comment](https://github.com/rails/rails/blob/a60e6ddb83b036f9b92af5dbcfda265cd0fde3b2/actionview/lib/action_view/template.rb#L155), it should be enough...